comps/integrations/haystack/src/opea_haystack/embedders/tei/text_embedder.py

```diff
@@ -0,0 +1,144 @@
+        """
+
+        self.api_url = api_url
+        self.prefix = prefix
+        self.suffix = suffix
+
+        if isinstance(truncate, str):
+            truncate = EmbeddingTruncateMode.from_str(truncate)
+        self.truncate = truncate
+
+        self.backend: Optional[Any] = None
+        self._initialized = False
+
+    def warm_up(self):
+        """
+        Initializes the component.
+        """
+        if self._initialized:
+            return
+
+        model_kwargs = {"input_type": "query"}
+        if self.truncate is not None:
+            model_kwargs["truncate"] = str(self.truncate)
+        self.backend = OPEABackend(
+            api_url=self.api_url,
+            model_kwargs=model_kwargs,
+        )
+
+        self._initialized = True
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            api_url=self.api_url,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            truncate=str(self.truncate) if self.truncate is not None else None,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "OPEATextEmbedder":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+            The deserialized component.
+        """
+        return default_from_dict(cls, data)
+
+    @component.output_types(embedding=List[float])
+    def run(self, inputs: Union[str, List[str]]):
+        """
+        Embed a string or list of strings.
+
+        :param inputs:
+            The text(s) to embed.
+        :returns:
+            A dictionary with the following keys and values:
+            - `embeddings` - Embeddngs of the text(s).
+        :raises RuntimeError:
+            If the component was not initialized.
+        :raises TypeError:
+            If the input is not a string or list of strings.
+        """
+        if not self._initialized:
+            msg = "The embedding model has not been loaded. Please call warm_up() before running."
+            raise RuntimeError(msg)
+        elif not isinstance(inputs, (str, list)):
+            msg = (
+                "OPEATextEmbedder expects a string or list of strings as an input."
+                "In case you want to embed a list of Documents, please use the OPEADocumentEmbedder."
+            )
+            raise TypeError(msg)
+        elif not inputs:
+            msg = "Cannot embed an empty string."
+            raise ValueError(msg)
+        
+        assert self.backend is not None
+        if isinstance(inputs, str):
+            inputs = [inputs]
+        text_to_embed = [self.prefix + text + self.suffix for text in inputs]
+        embedding = self.backend.embed(text_to_embed)
+
+        return {"emebddings": embedding}
```

Comment:
Optionally, meta data such as the model name or other info could be included here if you want. `return {"embedding": embedding, "meta": ...}`